tools/c7n_mailer/c7n_mailer/deploy.py

```diff
@@ -43,14 +44,16 @@ def get_archive(config):
         'c7n_mailer', 'ldap3', 'pyasn1', 'jinja2', 'markupsafe', 'ruamel',
         'redis', 'datadog', 'requests')
 
-    template_dir = os.path.abspath(
-        os.path.join(os.path.dirname(__file__), '..', 'msg-templates'))
+    for d in config['templates_folders']:
+        if not os.path.exists(d):
+            continue
+        for t in [f for f in os.listdir(d) if os.path.splitext(f)[1] == '.j2']:
+            with open(os.path.join(d, t)) as fh:
+                archive.add_contents('msg-templates/%s' % t, fh.read())
```

Comment:
what happens with duplicates when collapsing them into a single directory? i vaguely remember some issues with zip file for that case.

Reply:
Need to check this, but my expectation is second file overwrites first.
We can log warnings if there is a collision because this is definitely not an expected behavior and user needs to pay attention

Reply:
zip library shows a warning, so I think its ok to leave as-is (changed list of folder to the set)
```
/opt/rh/rh-python36/root/usr/lib64/python3.6/zipfile.py:1355: UserWarning: Duplicate name: 'mailer/msg-templates/default.html.j2'
  return self._open_to_write(zinfo, force_zip64=force_zip64)
```

Reply:
i think the actual manifest has two entries in that case afaicr, and the extract ends with last write wins

Reply:
What do you propose? 
I'm not sure if this is something that custodian needs to take care of. Warning is displayed to the user, so he should be able to fix duplicates issue.
I don't think that custodian should exit if it detects duplicates.. Its not an issue. Otherwise, its up to us picking first or second template.. I don't think there is any difference between them.